src/main/webapp/app/components/tabs/CurationToolsTab.tsx

```diff
@@ -1,105 +1,142 @@
-
-    if (results[0].ok && results[1].ok) {
-      // both succeeded
-      window.location.href = `${PAGE_ROUTE.CURATION}/${gene.name}`;
-    } else if (results[0].ok && !results[1].ok) {
-      // createMetaGene failed
-      notifyError(results[1].error);
-      deleteGene(genePath);
-    } else if (results[1].ok && !results[0].ok) {
-      // createGene failed
-      notifyError(results[0].error);
-      deleteMetaGene(metaGenePath);
-    } else {
-      // both failed
-      notifyError(new Error(`Errors: ${getErrorMessage(results[0].error)}, ${getErrorMessage(results[1].error)}`));
+  async function handleConfirmClick() {
+    const newGene = _.cloneDeep(geneToUpdate.current);
+
+    try {
+      const newFlag = (await searchFlags({ query: 'OncoKB' }))['data'].find(flag => isReleasedFlag(flag));
+      if (!newFlag) {
+        throw new Error('Error retrieving flag');
+      }
+
+      if (newGene.flags) {
+        newGene.flags.push(newFlag);
+      } else {
+        newGene.flags = [newFlag];
+      }
+      await updateGene(newGene);
+      await searchGenes({ query: gene.name }); // repopulate gene store entities
+    } catch (error) {
+      notifyError(error);
     }
   }
 
-  return (
-    <div>
-      <Row className="mb-2">
-        <Col className="px-0">
-          <h6>Curate New Gene</h6>
-        </Col>
-      </Row>
-      <Row className="mb-2">
-        <Col className="px-0">
-          <GeneSelect onChange={handleChangeSelectedGene} />
-        </Col>
-      </Row>
-      <Row className={`align-items-center ${showGeneExistsWarning ? 'justify-content-between' : 'justify-content-end'}`}>
-        {showGeneExistsWarning && (
-          <div className="gene-exists-warning mr-2">
-            <FaExclamationCircle className="mr-2" size={'25px'} color="danger" />
-            <span>Gene already exists</span>
-          </div>
-        )}
-        <Button color="primary" disabled={createButtonDisabled} onClick={handleCreateGene}>
-          Create
+  function getContent() {
+    if (isReleased === null) {
+      // loading state
+      return <></>;
+    }
+
+    if (isReleased) {
+      return (
+        <Row>
+          <FaRegCheckCircle size="24px" className="text-success mt-1 mr-2" />
+          <h6 style={{ marginTop: '6px' }}>Gene is released</h6>
+        </Row>
+      );
+    }
+
+    if (releaseGeneClicked) {
+      return (
+        <div>
+          <Row className="mb-1">
+            {getStatusIcon(!needsReview)}
+            Reviewed all content before releasing
+          </Row>
+          <Row className="mb-1">
+            {getStatusIcon(!geneSummaryEmpty)}
+            Gene summary is not empty
+          </Row>
+          <Row className="mb-1">
+            {getStatusIcon(!geneBackgroundEmpty)}
+            Gene background is not empty
+          </Row>
+          <Row className="mb-3">
+            {getStatusIcon(!geneTypeEmpty)}
+            Gene type is specified
+          </Row>
```

Comment:
Can we aggregate these checks into a list? The `Row` here can be populated based on the list. Also `confirmButtonDisabled` can be calculated based on the list. 
Test needs to be differentiated with `optional` and `requried`